pkg/kubernetes/node/v1alpha1/kubernetes.go

```diff
@@ -165,6 +204,31 @@ func (k *Kubeclient) Get(name string, opts metav1.GetOptions) (*corev1.Node, err
 	return k.get(cli, name, opts)
 }
 
+// GetHostName returns a hostname corresponding
+// to the provide node name
+func (k *Kubeclient) GetHostName(name string, opts metav1.GetOptions) (string, error) {
```

Comment:
Should we fallback and use the node name, incase it failed to find a hostname for the node?

Reply:
Makes sense to return the name as hostname if the hostname is not set on the nodes. ( cc: @AmitKumarDas )

Reply:
Isn't this always set by K8s? @akhilerm @kmova

Reply:
done